qualang_tools/plot/plot.py

```diff
@@ -170,77 +170,146 @@ def plot_demodulated_data_1d(
     return fig
 
 
-def get_simulated_samples_by_element(element_name: str, job: QmJob, config: dict):
+def get_simulated_samples_by_element(
+    element_name: str, job: QmJob, config: dict
+) -> Tuple:
     """
-    Extract the simulated samples corresponding to a given element.
+    Extract the simulated samples (analog and digital) corresponding to a given element.
 
     :param element_name: name of the element corresponding to the samples to extract. Must be defined in the config.
     :param job: The simulated QmJob from which the samples will be extracted.
     :param config: The config file used to create the job.
-    :return: 1D complex array containing the simulated samples for the specified element.
+    :return: Two 1D lists containing the simulated analog and digital samples for the specified element.
     """
     element = config["elements"][element_name]
     job.result_handles.wait_for_all_values()
     sample_struct = job.get_simulated_samples()
+    # Analog waveforms
     if "mixInputs" in element:
         port_i = element["mixInputs"]["I"]
         port_q = element["mixInputs"]["Q"]
-        samples = (
+        analog_samples = (
             sample_struct.__dict__[port_i[0]].analog[str(port_i[1])]
             + 1j * sample_struct.__dict__[port_q[0]].analog[str(port_q[1])]
         )
-    else:
+    elif "singleInput" in element:
         port = element["singleInput"]["port"]
-        samples = sample_struct.__dict__[port[0]].analog[str(port[1])]
-    return samples
+        analog_samples = sample_struct.__dict__[port[0]].analog[str(port[1])]
+    else:
+        analog_samples = None
+    # Digital waveforms
+    digital_samples = []
+    if "digitalInputs" in element:
+
+        for key in element["digitalInputs"].keys():
+            port = element["digitalInputs"][key]["port"]
+            digital_samples.append(
+                sample_struct.__dict__[port[0]].digital[str(port[1])].astype(int)
+            )
+    else:
+        digital_samples.append(None)
+
+    return analog_samples, digital_samples
 
 
 def plot_simulator_output(
-    plot_axes: List[List[str]], job: QmJob, config: dict, duration_ns: int
+    job: QmJob,
+    config: dict,
+    duration_ns: int,
+    plot_axes: List[List[str]] = (()),
+    **kwargs,
 ):
     """
     Generate a 'plotly' plot of simulator output by elements
 
-    :param plot_axes: a list of lists of elements (ex: [["qubit0"], ["qubit1"]]). Will open
-    multiple axes, one for each list.
+    :param plot_axes: (optional) a list of lists of elements (ex: [["qubit0", "qubit1"], ["resonator0", "resonator1"]]). Will open
+    multiple axes, one for each list. If not provided, then the qua_program must be specified as kwargs using qua_program= and all the elements will be plotted in their own axis.

```

Comment:
Very confusing to have two optional, one in kwargs...
Why not add `qua_program` to the variables as well?

Reply:
The thing is we don't need the QUA program except when plot_axes isn't provided and I understand that it can be painful for the user to input the qua program every time...
I tried to get the elements used in the program from the job (instead of the program) but I didn't manage...

Reply:
Look at the comment I added.
and then here:
```suggestion
    :param plot_axes: (Either this or 'program' have to be provided) a list of lists of elements (ex: [["qubit0", "qubit1"], ["resonator0", "resonator1"]]). Will open multiple axes, one for each list. 
    :param qua_program: (Either this or 'plot_axes' have to be provided) All the elements will be plotted in their own axis.

```

What does it mean that everyone will be in its own axis? different plots?

Reply:
It's a request from a customer actually

Reply:
seems a bit extreme to me, but ok.

Reply:
Why extreme?

Reply:
Maybe bloat is a better word?

To me, it seems more friendly to have all of the elements on the same axis. It's hard for me to imagine wanting them on seperate axes, it would take too much space and would be hard to compare and check timing.

A much better option, I think, is having them on the same axis but shifted in the Y axis to be one on top of the other, this is how usually timing is displayed.

Reply:
I see, I guess it is for readability issue if you have a lot of elements... And note that with plotly comparing different axes is much better than with matplotlib and I kind of like it this way, you should try it.